### ISSUE TITLE
Hardcoded namespace breaks `agones-controller` deployment in non-default namespaces

### ISSUE THREAD

**Issue by maxcorbin** [googleforgames/agones#3208]
# Expected
Upgrading to `>=1.31.0` should be backwards compatible with existing deployments.

# Actual
Upgrading to `>=1.31.0` causes the controller to be unresponsive to API calls in deployments outside of namespaces named `agones-system`. This prevents fleets from autoscaling or from replacing shutdown gameservers.

# Description
In `main.go` for the `agones-controller` deployment, there is a hard-coded namespace argument that is set to `agones-system`. While I know that this is the recommended setup, the helmchart does accept other namespaces and will deploy just fine into `default`, for example. Such deployments are broken on `>=1.31.0` due to the new `SplitControllerAndExtensions` work. [I believe the problem is here.](https://github.com/googleforgames/agones/blob/9840265f3855f7db143e28ab3c14ad5431fe6572/cmd/controller/main.go#L465)

# Diagnosis
The issue can be diagnosed by the presence of this in the controller logs:

```
leaderelection.go:334] error initially creating leader election record: namespaces "agones-system" not found
```

# Attempted Solutions
I tried scaling down to 1 replica and I also tried turning the `SplitControllerAndExtensions` flag off in the hopes that it would avoid the codepath where the hardcoded value is, but this doesn't work. I believe that the namespace value is not gated behind any control flow or other logic, and that it will affect all deployments that use a non-default namespace for versions 1.31.0 and on, even when leader election is not needed.

# Workaround
We resolved this in the short term by rolling back to `1.30.0`. Obviously we would prefer not to stay on the older version for too long, but we are unable to upgrade until this is fixed. I don't see an easy way to refer to the namespace that is provided to the deployment in the Helm chart, or else I would be happy to provide a PR to fix the problem.

Please let me know if there's additional information needed, or if I am missing something simple that would help me write a PR.

**Comment by markmandel** [comment]
That's a legit bug. Thanks for filing the issue!

**Comment by zmerlynn** [comment]
Fixed by #3209 - thanks for the report!
